File: 2019.03.17/python/main.py
def findLowestPositive(numList): 
    # get min and max num
    min_num = min(numList)
    max_num = max(numList)

    # iterate from min_num to max_num incrementing by 1
    for num in range(min_num, max_num+1):
        if not num in numList and num > 0:
            return num
        elif num == max_num:
            return num+1

numList1 = [3, 4, -1, 1]
numList2 = [1, 2, 0]

result1 = findLowestPositive(numList1)
result2 = findLowestPositive(numList2)

print(numList1, " ->", result1)
print(numList2, " ->", result2)

File: 2019.03.17/c++/main.cpp
#include<iostream>
#include<vector>
#include<algorithm>

using namespace std;

int findLowestPositive(vector<int> const &numList) {
    // find min and max value
    int min = *min_element(numList.begin(), numList.end());
    int max = *max_element(numList.begin(), numList.end());

    // iterate from min to max by an increment of 1
    int num;
    for (num = min; num <= max; num++) {
        // check if num is not on numList
        for(auto i = numList.begin(); i != numList.end(); i++) {
            if (num == *i) {
                goto continue_out; 
            }
        }
        // check if num is a positive
        if (num > 0) return num;
        continue_out:;
    }
    return num;
}


int main() {
    vector<int> numList1 {3, 4, -1, 1};
    vector<int> numList2 {1, 2, 0};

    int result1 = findLowestPositive(numList1);
    int result2 = findLowestPositive(numList2);

    cout << "result1: " << result1 << endl;
    cout << "result2: " << result2 << endl;

    return 0;
}